mysqlnd_azure.c

```diff
@@ -674,6 +711,43 @@ MYSQLND_METHOD(mysqlnd_azure, connect)(MYSQLND * conn_handle,
 }
 /* }}} */
 
+/* {{{ mysqlnd_azure_apply_resources, do resource apply works when module init*/
+int mysqlnd_azure_apply_resources() {
+  if (MYSQLND_AZURE_G(logOutput) == 2 && MYSQLND_AZURE_G(logLevel) > 0) {
+    char *logfilePath = NULL;
+    int logflag = 0;
+    if (ZSTR_LEN(MYSQLND_AZURE_G(logfilePath)) > 155) {
+      logflag = 1;
+      logfilePath = "mysqlnd_azure_runtime.log";
+    } else {
+      logfilePath = ZSTR_VAL(MYSQLND_AZURE_G(logfilePath));
+    }
+
+    OPEN_LOGFILE(logfilePath);
+
+    if (logflag) {
+      php_error_docref(NULL, E_WARNING, "[mysqlnd_azure] Logfile string length exceeds 255, redirected to mysqlnd_azure_runtime.log");
+      AZURE_LOG_SYS("Given logfile name too long, redirected to default: mysqlnd_azure_runtime.log");
+    }
+
+    if (logfile == NULL) {
+      php_error_docref(NULL, E_WARNING, "[mysqlnd_azure] Unable to open log file: %s", logfilePath);
+      return 1;
+    }
+  }
+  return 0;
+}
+/* }}} */
+
+/* {{{ mysqlnd_azure_release_resources, release resources when module destruct */
+int mysqlnd_azure_release_resources() {
+  if (MYSQLND_AZURE_G(logOutput) == 2 && MYSQLND_AZURE_G(logLevel) > 0 && logfile) {
```

Comment:
we might could always call CLOSE_LOGFILE?

Reply:
never mind, just in case someone made more code changes in future and miss to change this code line.

Reply:
Updated.
[+] Add some comments on code itself (resource_apply/ resource_release).
[m] open/close file handler when configured print logs to file
       logFile always opened if configured log to file, whatever the logLevel value is.

---
Former implementation may trigger some misleading behavior:
file opened but no close (if user change logLevel: 0 -> $some_number)
module exit without file close (if user change logLevel: $some_nuber -> 0)